README.md

```diff
@@ -43,12 +48,94 @@ In your terminal, navigate to `<your-project>/wp-content/themes/flynt-starter-th
+
+For example, take `templates/page.php`:
+
+```php
+<?php
+
+Flynt\echoHtmlFromConfigFile('default.json');
+```
+
+The JSON template configuration files are found in `config/templates`. These configuration files define the [areas](#areas) and [components](#components) which are loaded into the template.
+
+Take `config/templates/default.json` as an example. This template calls the `DocumentDefault` component and registers one area: `layout`. The `layout` area calls the `LayoutSinglePost` component, which in turn contains three nested [areas](#areas): `mainHeader`, `pageComponents`, and `mainFooter`. In addition, the `pageComponents` area calls the `ComponentLoaderFlexible` component.
+
+```json
+{
+  "name": "DocumentDefault",
+  "areas": {
+    "layout": [
+      {
+        "name": "LayoutSinglePost",
+        "areas": {
+          "mainHeader": [],
+          "pageComponents": [
+            {
+              "name": "ComponentLoaderFlexible",
+              "customData": {
+                "fieldGroup": "pageComponents"
+              }
+            }
+          ],
+          "mainFooter": []
+        }
+      }
+    ]
+  }
+}
+```
+
+To complete the picture, the `layout` area defined in the template is then output in `Components/DocumentDefault/index.twig`:
+
+```twig
+<!DOCTYPE html>
+<html class="flyntComponent {{ body_class }}" lang="{{ site.language }}" dir="{{ dir }}" is="flynt-document-default">
+  <head><!--...--></head>
+  <body role="document">
+    <!-- Output the `layout` area as defined in `config/templates/default.json`. -->
+    {{ area('layout') }}
+    {{ wp_footer }}
+  </body>
+</html>
+```
+
+### Components
+A component is a self-contained building-block. As such, each component is kept within its own folder which contains everything it requires; the layout, the ACF field setup, all necessary WordPress filter and hook logic, scripting, styles, and images.
+
+```
+  ExampleComponent/
+  ├── assets/
+  |   ├── exampleImage.jpg
+  |   └── exampleIcon.svg
+  ├── fields.json
+  ├── functions.php
+  ├── index.twig
+  ├── README.md
+  ├── script.js
+  ├── style.styl
+```
+
+Building components is a sustainable process, meaning every component you develop can be reused within a project, or in another; increasing your head-start with every new Flynt project.
+
+### Areas
+Since components are self-contained, areas provide a way to stack our building-blocks together. An area is simply a location within a component where it is possible to add other components.
+
+### Features
+With WordPress, it is easy to create one large `functions.php` file, crammed full of all the custom logic your theme may need. This can get messy. In Flynt, we split each piece of functionality into smaller, self-contained **feature** bundles.
+
+In most cases, features add global hooks and filters that affect the project on a global level. With this in mind, each feature is built with drag and drop reusability in mind.
```

Comment:
`drag & drop reusability` reminds me of backend d&d feature. just `reusability`?